GitUI/CommandsDialogs/FormEditor.cs

```diff
@@ -116,40 +116,60 @@ private void FormEditor_FormClosing(object sender, FormClosingEventArgs e)
+        private void SaveChanges(bool rethrowException = false)
         {
-            if (!string.IsNullOrEmpty(_fileName))
+            try
             {
-                if (fileViewer.FilePreamble is null || Module.FilesEncoding.GetPreamble().SequenceEqual(fileViewer.FilePreamble))
-                {
-                    File.WriteAllText(_fileName, fileViewer.GetText(), Module.FilesEncoding);
-                }
-                else
+                if (!string.IsNullOrEmpty(_fileName))
                 {
-                    using (var bytes = new MemoryStream())
+                    if (fileViewer.FilePreamble is null || Module.FilesEncoding.GetPreamble().SequenceEqual(fileViewer.FilePreamble))
                     {
-                        bytes.Write(fileViewer.FilePreamble, 0, fileViewer.FilePreamble.Length);
-                        using (var writer = new StreamWriter(bytes, Module.FilesEncoding))
+                        using (FileStream fs = new FileStream(_fileName, FileMode.Open))
                         {
-                            writer.Write(fileViewer.GetText());
+
+                            using (TextWriter tw = new StreamWriter(fs, Module.FilesEncoding, bufferSize: 1024,
+                                leaveOpen: true))
+                            {
+                                tw.Write(fileViewer.GetText());
+                            }
+
+                            // after flushing, set the stream length to the current position in order to truncate leftover text
+                            fs.SetLength(fs.Position);
                         }
+                    }
+                    else
+                    {
+                        using (var bytes = new MemoryStream())
+                        {
+                            bytes.Write(fileViewer.FilePreamble, 0, fileViewer.FilePreamble.Length);
+                            using (var writer = new StreamWriter(bytes, Module.FilesEncoding))
+                            {
+                                writer.Write(fileViewer.GetText());
+                            }
 
-                        File.WriteAllBytes(_fileName, bytes.ToArray());
+                            File.WriteAllBytes(_fileName, bytes.ToArray());
+                        }
                     }
+
+                    // we've written the changes out to disk now, nothing to save.
+                    HasChanges = false;
+                }
+            }
+            catch (Exception ex)
+            {
+                if (rethrowException)
+                {
+                    throw;
                 }
 
-                // we've written the changes out to disk now, nothing to save.
-                HasChanges = false;
+                MessageBox.Show(this, _cannotSaveFile.Text + Environment.NewLine + ex.Message, Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
```suggestion
                MessageBox.Show(this, $"{_cannotSaveFile.Text}{Environment.NewLine}{ex.Message}", Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Reply:
I copied this when I moved it. Depending on the comment below, would you like it changed if the try/catch move is reverted?I'd be adding it to the other location did not have a try/catch in that case as well.